docs/docs/manual/exporting.md

```diff
@@ -6,34 +6,76 @@ sidebar_label: Exporting
 *   ODF spreadsheet
-*   Google Sheets \
+*   Upload to Google Sheets (requires Google account authorization)
+*   [Custom tabular exporter](#custom-tabular-exporter)
+*   [SQL statement exporter](#sql-statement-exporter)
+*   [Templating exporter](#templating-exporter)
 
-*   Custom tabular export
-*   SQL
-*   Templating export \
+You can also [export reconciled data to Wikidata](wikidata#editing-wikidata-with-openrefine), or save your schema for future use with others OpenRefine projects:
 
 *   Upload edits to Wikidata
 *   Export to QuickStatement
 *   Export Wikidata schema
 
+### Custom tabular exporter
+
+![A screenshot of the custom tabular content tab.](/img/custom-tabular-exporter.png)
+
+With the custom tabular exporter, you can choose which of your data to export, the separator you wish to use, and whether you'd like to download it to your computer or upload it into a Google Sheet. This exporter is especially useful with reconciled data, as you can choose whether you wish to output the cells' original values, the matched values, or the matched IDs. 
+
+You can also choose how to output date-formatted cells, and whether you wish to export all the project's data or just the current view set. 
+
+![A screenshot of the custom tabular file download tab.](/img/custom-tabular-exporter2.png)
+
+With the download options, you can generate a preview of how the first ten rows of your dataset will output. If you do not choose one of the file formats on the right, the "Download" button will generate a text file. 
+
+With the "Option Code" tab, you can copy JSON data that saves your current settings, to reuse on another project, or you can paste in existing JSON settings data to apply to the current project. 
+
+### SQL statement exporter
+
+The SQL statement exporter creates a table containing the data you’ve exported, which you can use to overwrite an existing database.
+
+![A screenshot of the SQL statement creator window.](/img/sql-exporter.png)
+
+A window with two tabs will pop up: one to define what data to output, and another to modify the statement table. This allows you to craft a complete SQL command, including DROP and IF EXISTS if you require them. 
+
+![A screenshot of the SQL statement download window.](/img/sql-exporter2.png)
+
+
+### Templating exporter
+
+If you pick <span class="menuItems">Templating…</span> from the <span class="menuItems">Export</span> dropdown menu, you can “roll your own” exporter. This is useful for formats that we don't support natively yet, or won't support. For example, the Templating exporter generates JSON by default. 
+
+The window that appears allows you to set your own separators, prefix, and suffix to create a complete dataset in the language of your choice. In the Row Template section, you can choose which columns to generate from each row by calling them with variables. 
+
+![A screenshot of the Templating exporter generating JSON by default.](/img/templating-exporter.png)
+
```

Comment:
I think it would be worth expanding a bit here, on the following topics:
* the syntax of the templates (how do you introduce a template bit? how do you escape braces?)
* the evaluation context of the expressions used in the template: what variables can you use there?
* can you use other expression languages than GREL inside the templates?

Reply:
Thanks for these suggestions. I'm able to expand on the variables and some syntax, but: _can_ you use other expression languages? I wouldn't know where to begin figuring that out on my own. Is there a jsonize() equivalent in clojure? I can't seem to get any clojure syntax to work but that's hardly an indicator of anything.

Reply:
My mistake, you've tricked me. How _do_ you escape a curly brace? Our regular approach is useless here.

Reply:
Good points, I don't actually know the answer to any of these questions! I'll investigate in the code and report back.

Reply:
So, looking at the code (I haven't checked that it does work, it'd be useful if you do)
- The characters `\`, `{` and `$` can be escaped with `\` (so, `\\`, `\{` and `\$` respectively produce `\`, `{` and `$`).
- You can use `${myColumn}` to insert values from a given column
- You can use `{{ some expression }}` to insert values obtained by any expression. By default the expression is expected to be written in GREL, but you can use other languages too: `{{jython:return cells["myColumn"].value}}` should work as well (and same for clojure).
It's not clear to me if there is a simple way for users to figure out the exact language identifier (here `jython`) to use - it can be observed in the history when you use these expressions in other operations, but that is not very convenient. Perhaps we could just list the available language identifiers in this part of the documentation for convenience? Or in the general documentation about expressions if this is useful elsewhere.

Reply:
I don't have any luck escaping curly braces in regex with a backslash. 
`value.match(/.*(\d\d\d\d)/)` works fine but `value.match(/.*(\d\{4\})/)` does not resolve - it just outputs the entire {{}} formula as text.

The statement `value.find(/[,.!{@#$\?]/)` doesn't need the opening curly brace escaped. But it won't accept a CLOSING curly brace, with or without a backslash. 
I don't seem to need to escape a $ at all - the GREL runs just fine with it in. I don't need to cancel a backslash, but I do need to cancel a forward slash with a backslash. (`value.find(/[,.!{@#\/$\?]/)`)
So, I am not sure what this has to do with the code you looked at. Help please!


Thanks for the jython though!

Reply:
I would not be surprised if this escaping code was buggy. Let's debug this together in today's meeting perhaps?

Reply:
> Anything that appears inside curly braces ({{}}) is treated as a GREL expression; anything outside is generated as straight text. You can use Jython or Clojure by declaring it at the start: for example, {{jython:return cells["Author"].value}} will run a Jython expression.
> 
> caution
> 
> Note that some syntax is different in this tool than elsewhere in OpenRefine: a forward slash must be escaped with a backslash, while other characters do not need escaping. You cannot, at this time, include a closing curly brace (}) anywhere in your expression, or it will cause it to malfunction.

Here's what we've got for now.

Reply:
Is it not worth mentioning the `${columnName}` syntax too?

Reply:
It is, I totally forgot. Thanks!